### PR TITLE
Rat trap patch

### PR DESCRIPTION
**Pull request by Fusselwurm**
@nomisum not sure which changes are sensible here, which should be specific to rat trap etc.

### CODE REVIEW

File: Custom_GruppeAdler/createVehicle.sqf [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+
+	_createDriver = {
+		_grp0 = _this select 0;
+		_car0 = _this select 1;
+
+		_crewDriver = _grp0 createUnit ["C_man_1", _pos, [], 0, "CARGO"];
+
+		[_crewDriver] call randomCivilian; // rebel loadout
+		[_crewDriver, _car0, true] call ACE_VehicleLock_fnc_addKeyForVehicle;
```
Comment (nomisum) [inline Custom_GruppeAdler/createVehicle.sqf:24]: gives the driver a key for the car - but car is *locked* initially, so driver has to get out and ripped to drive it.

File: Custom_GruppeAdler/createVehicle.sqf [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+	_veh = createVehicle [_vehicleType, _pos, [], 0, "NONE"];
+	waitUntil {!isNull _veh};
+
+	_group = createGroup _side;
+
+
+	[_group,_veh] call _createDriver;
+
+	if (_vehicleType == "RDS_Ikarus_Civ_01" || _vehicleType == "RDS_Ikarus_Civ_02") exitWith {
```
Comment (nomisum) [inline Custom_GruppeAdler/createVehicle.sqf:56]: ikarus can have some more cargo people

File: Custom_GruppeAdler/createVehicle.sqf [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	};
+
+
+	if (_crewCount > 0) then {
+		for [{_i=0}, {_i<_crewCount}, {_i=_i+1}] do {
+		[_group,_veh] call _createCargo;
+		};
+	};
+
+	_veh addEventhandler ["HandleDamage",{if ((_this select 4) == "") then {0};}]; // ignore crash damage
```
Comment (nomisum) [inline Custom_GruppeAdler/createVehicle.sqf:73]: vehicles dont take damage if damage source (projectile!) is unknown

File: Custom_GruppeAdler/streetsTanoa.sqf [hunk 4]
```diff
@@ -0,0 +1,14015 @@
...
+1680635,
+1680730,
+1678196,
+1681564,
+1681491,
+1677491,
+1677490,
+1677489,
+1677525,
+1677524
```
Comment (nomisum) [inline Custom_GruppeAdler/streetsTanoa.sqf]: tanoa streets are not necessary at all. are not used in any form and probably never

File: Init.sqf [hunk 5]
```diff
@@ -17,3 +17,15 @@ if (isServer) then {
 	call compile preprocessFileLineNumbers "node_modules\engima-traffic\Server\Functions.sqf";
 	call compile preprocessFileLineNumbers "node_modules\engima-traffic\ConfigAndStart.sqf";
 };
+
+
+setCustomFace =
```
Comment (nomisum) [inline Init.sqf]: not used afaik, but handy if you want a specific ethnicity

File: setVehicleLightHit.sqf [hunk 6]
```diff
@@ -0,0 +1,21 @@
+private ["_object","_sPos","_tPos","_bullet","_bPos","_heading","_velocity"];
```
Comment (nomisum) [inline setVehicleLightHit.sqf]: afaik didnt work

File: rat-trap.patch [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+ 	"RDS_Lada_Civ_03",
+-	"RDS_Lada_Civ_04",
++
+ 	"RDS_Lada_Civ_05",
++	"RDS_Lada_Civ_01",
+ 	"RDS_Lada_Civ_02",
+ 	"RDS_Lada_Civ_03",
+-	"RDS_Lada_Civ_04",
+ 
+-	"RDS_Zetor6945_Base",
```
Comment (nomisum) [inline rat-trap.patch]: tractor?

File: rat-trap.patch [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+-	"RHS_Ural_Open_Civ_03",
+-	"rhsgref_nat_uaz"
++	"RHS_Ural_Open_Civ_03"
+ 	]],
+ 	["VEHICLES_COUNT", 10],
+-	["MIN_SPAWN_DISTANCE", 1500],
+-	["MAX_SPAWN_DISTANCE", 2500],
+-	["MIN_SKILL", 1],
+-	["MAX_SKILL", 1],
++	["MIN_SPAWN_DISTANCE", 800],
```
Comment (nomisum) [inline rat-trap.patch]: 800 is quite close for minimum spawn distance. probably enough if no air is involved

File: rat-trap.patch [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
++                };
+ 
+                 if (_areaMarkerName != "" && !([getPos _roadSegment, _areaMarkerName] call ENGIMA_TRAFFIC_PositionIsInsideMarker)) then {
+                 	_insideMarker = false;
+@@ -250,7 +294,7 @@ ENGIMA_TRAFFIC_StartTraffic = {
+ 
+ 	            _isOk = true;
+ 
+-	            if (_tooClose || _tooFarAwayFromAll || _tooCloseToAnotherVehicle || !_insideMarker) then {
++	            if (_tooClose || _tooFarAwayFromAll || _tooCloseToAnotherVehicle || !_insideMarker || !_isRoad) then {
```
Comment (nomisum) [inline rat-trap.patch]: the whole concept of checking for roads didnt work out in the end, as waypoints/driving routes are set by the engine. now its better to use the new driving ai commands anyway

File: Custom_GruppeAdler/createVehicle.sqf (hunk 3 above)
Comment (Fusselwurm) [reply Custom_GruppeAdler/createVehicle.sqf:73]: wot. wir erschaffen uncrashbare zombiefahrzeuge? hmmm :/ klingt nach irgendwas das in ne config sollte, oder von außen gesetzt wird.

Comment (nomisum) [reply Custom_GruppeAdler/createVehicle.sqf:73]: driving KI führt sonst dazu, dass je nach map sehr schnell alle irgendwo am straßenrand liegen.

File: Init.sqf (hunk 5 above)
Comment (Fusselwurm) [reply Init.sqf]: not core functionality, can be added by spawn event handler, will remove it.

File: rat-trap.patch (hunk 7 above)
Comment (Fusselwurm) [reply rat-trap.patch]: uh ka. würd die config eh noch iwie auslagern wollen, die is kacke so

File: rat-trap.patch [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+ 	["VEHICLES_COUNT", 10],
+-	["MIN_SPAWN_DISTANCE", 1500],
+-	["MAX_SPAWN_DISTANCE", 2500],
+-	["MIN_SKILL", 1],
+-	["MAX_SKILL", 1],
++	["MIN_SPAWN_DISTANCE", 800],
++	["MAX_SPAWN_DISTANCE", 2000],
++	["MIN_SKILL", 0.4],
++	["MAX_SKILL", 0.6],
++  ["AREA_MARKER", "mrk_drive_area"],
```
Comment (Fusselwurm) [inline rat-trap.patch]: ist der pflicht, btw?

File: rat-trap.patch (hunk 9 above)
Comment (Fusselwurm) [reply rat-trap.patch]: will remove _isRoad check

**Pull request merged**